pymongo/auth_shared.py

```diff
@@ -87,6 +87,16 @@ def __hash__(self) -> int:
 """Mechanism properties for MONGODB-AWS authentication."""
 
 
+def _validate_canonicalize_host_name(value: str | bool) -> str | bool:
+    valid_names = [False, True, "none", "forward", "forwardAndReverse"]
```

Comment:
So "none", "forward", and "forwardAndReverse" are all supposed to be treated the same as True? Is that accurate?

Reply:
No, you'll see in the follow up PR how they're used in practice.